### PR TITLE
[full-ci] enhancement: handle ocm event notifications

### PR DESCRIPTION
**Pull request by fschade**
## Description
By now, the sciencemesh part in reva was responsible to notify the invitee that a invitaion happened, in ocis we have a event driven approach.

This PR introduces that and handles those invitation events in the notification service by sending mails.

## Related Issue
- Fixes https://github.com/owncloud/ocis/issues/9583
- Needs: https://github.com/cs3org/reva/pull/4832
- Needs: https://github.com/owncloud/web/pull/11512

## How Has This Been Tested?
- not at all right now, next step

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by fschade** [comment]
@micbar what about our other services (activitylog, audit, ...), should i add the new event there too?

**Comment by micbar** [comment]
> @micbar what about our other services (activitylog, audit, ...), should i add the new event there too?

Yes, that would be awesome!

File: services/notifications/pkg/email/templates.go [hunk 1]
```diff
@@ -72,18 +72,39 @@ You might still have access through your other groups or direct membership.`),
...
+
+	ScienceMeshInviteTokenGenerated = MessageTemplate{
+		textTemplate: "templates/text/email.text.tmpl",
+		htmlTemplate: "templates/html/email.html.tmpl",
+		// ScienceMeshInviteTokenGenerated email template, Subject field (resolves directly)
+		Subject: l10n.Template(`ScienceMesh: {InitiatorName} wants to collaborate with you`),
+		// ScienceMeshInviteTokenGenerated email template, resolves via {{ .Greeting }}
+		Greeting: l10n.Template(`Hi,`),
+		// ScienceMeshInviteTokenGenerated email template, resolves via {{ .MessageBody }}
+		MessageBody: l10n.Template(`{ShareSharer} ({ShareSharerMail}) wants to start sharing OCM resources with you.
```
Comment (mmattel) [inline services/notifications/pkg/email/templates.go:84]:
```suggestion
		MessageBody: l10n.Template(`{ShareSharer} ({ShareSharerMail}) wants to start sharing collaboration resources with you.
```
This is user facing, IMHO, no one understands OCM...

File: services/notifications/pkg/email/templates.go [hunk 2]
```diff
@@ -72,18 +72,39 @@ You might still have access through your other groups or direct membership.`),
...
+{{if .ShareLink }}To accept the invite, please visit the following URL:
+{ShareLink}
+
+Alternatively, you can{{else}}
+Please{{end}} visit your mesh provider and use the following details:
```
Comment (mmattel) [inline services/notifications/pkg/email/templates.go:89]:
```suggestion
Please{{end}} visit your federation provider and use the following details:
```
Mesh, while technically correct, is hard to understand for end users. While, imho, federation is more clear and will be used in the docs.

**Comment by mmattel** [comment]
@fschade It seems that the audit services also contains translation relevant strings. You therefore need to have a look at https://owncloud.dev/services/general-info/add-translations/ for a how to. This includes the translation setup and the readme addition.

**Review by micbar — APPROVED** [review]
Lgtm!

**Review by butonic — APPROVED** [review]

**Pull request merged**